testing/src/test/java/org/oppia/android/testing/math/PolynomialSubjectTest.kt

```diff
@@ -0,0 +1,173 @@
+package org.oppia.android.testing.math
+
+import org.junit.Assert.assertThrows
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+import org.oppia.android.app.model.Polynomial
+import org.oppia.android.app.model.Real
+
+/** Tests for [PolynomialSubject]. */
+@RunWith(JUnit4::class)
+class PolynomialSubjectTest {
+
+  @Test
+  fun testPolynomialSubject_withNullPolynomial_isNotValidPolynomial() {
+    PolynomialSubject.assertThat(null).isNotValidPolynomial()
+  }
+
+  @Test
+  fun testPolynomialSubject_withNonNullPolynomial_isNotValidPolynomial_fails() {
+    val polynomial = Polynomial.newBuilder()
+      .addTerm(Polynomial.Term.newBuilder().setCoefficient(Real.newBuilder().setInteger(1)))
+      .build()
+    assertThrows(AssertionError::class.java) {
+      PolynomialSubject.assertThat(polynomial).isNotValidPolynomial()
+    }
+  }
+
+  @Test
+  fun testPolynomialSubject_withConstantPolynomial_isConstantThat() {
+    val constantPolynomial = Polynomial.newBuilder()
+      .addTerm(Polynomial.Term.newBuilder().setCoefficient(Real.newBuilder().setInteger(5)))
+      .build()
+    PolynomialSubject.assertThat(constantPolynomial)
+      .isConstantThat()
+      .isIntegerThat()
+      .isEqualTo(5)
+  }
+
+  @Test
+  fun testPolynomialSubject_withNonConstantPolynomial_isConstantThat_fails() {
+    val nonConstantPolynomial = Polynomial.newBuilder()
+      .addTerm(
+        Polynomial.Term.newBuilder()
+          .setCoefficient(Real.newBuilder().setInteger(5))
+          .addVariable(Polynomial.Term.Variable.newBuilder().setName("x").setPower(1))
+      )
+      .build()
+    assertThrows(AssertionError::class.java) {
+      PolynomialSubject.assertThat(nonConstantPolynomial).isConstantThat()
+    }
+  }
+
+  @Test
+  fun testPolynomialSubject_withZeroTerms_hasTermCountThat() {
```

Comment:
```suggestion
  fun testPolynomialSubject_withZeroTerms_hasTermCountThatIsEqualToZero() {
```

Explicitly names the expected outcome.